### ISSUE TITLE
Building-Type Quest: This ist a building:part

### ISSUE THREAD

**Issue by DeusFigendi** [streetcomplete/StreetComplete#3351]
In the area where I live there are many building parts that a mapped as own buildings. And of coarse StreetComplete asks for every "building" what type (and number of levels and roof shape) it is/has.

I would **love** to have an answer like *"this is a building:part not an own building"* as "other answers".

For example [this building](https://www.openstreetmap.org/way/394608124) is in fact a building part I think, as one might see in [this phot](https://duckduckgo.com/?q=Lage&t=hy&va=g&ia=web&iaxm=maps&strict_bbox=1&bbox=52.027884345268006%2C8.769776330100683%2C52.02641555065364%2C8.77385328780332)[ography](https://www.bing.com/maps?osid=3180e573-0ab8-4f2c-a48f-01f6d233bc46&cp=52.027308~8.770973&lvl=19&style=h&v=2&sV=2&form=S00027)

The best outcome for me would be if I could choose "this is a building:part" and streetComplete asks me of what building this is a part of. And then extends the chosen building to the form of the building I was asked for and makes that one to a bulding:part

Okay that was complicated, I'll draw something…
![Rastergrafik](https://user-images.githubusercontent.com/1711504/136633093-d91680b8-31f9-4b35-8199-4697fc8c3125.png)

Well, maybe this is too complicated (the extension of the building) in that case it would also be fine to change the building to a building:part and leave kinda fixme that says "this is part of another building (with ID 123456) please join those".
And maybe this is a very special problem that only occurs in my region but in occurs massively here!

**Comment by westnordost** [comment]
The UI could work and it would be possible to implement. However, it is quite some work. Especially the edge cases like

![image](https://user-images.githubusercontent.com/4661658/136664873-f3da207d-dcc7-460a-b15c-02cc2804f27b.png)

will complicate this a lot. So, I will not do this, too bad ratio between gain and effort.

In case you want to do this or someone else reading this, the structure can be similar to split ways. 
See 
- data structure to persist the (intention of the) split before it is being applied: https://github.com/streetcomplete/StreetComplete/blob/master/app/src/main/java/de/westnordost/streetcomplete/data/osm/edits/split_way/SplitPolylineAtPosition.kt
- code that does the split: https://github.com/streetcomplete/StreetComplete/blob/master/app/src/main/java/de/westnordost/streetcomplete/data/osm/edits/split_way/SplitWayAction.kt
- UI for adding the split-points: https://github.com/streetcomplete/StreetComplete/blob/master/app/src/main/java/de/westnordost/streetcomplete/quests/SplitWayFragment.kt

**Comment by RubenKelevra** [comment]
@westnordost I understand that a "full fix" is impossible. But does this mean that there should not be a partial fix?

I mean if there are two simple rectangulars with two common nodes SC could easily merge them into a building and make the building:part thing itself.

If that's not possible for SC it could create an note, with both buildings mentioned and ask the user to provide further evidence etc.

Currently there's no such thing in the whole UI - so the users of SC don't know nothing about building parts. This is by far *the worst* "solution", as the users will just create more info on the "secondary building" and even try to find the house number and mark that it doesn't have one etc.

**Comment by matkoniecz** [comment]
> I mean if there are two simple rectangulars with two common nodes SC could easily merge them into a building and make the building:part thing itself.

> If that's not possible for SC it could create an note, with both buildings mentioned and ask the user to provide further evidence etc.

Both will result in confused users why it works only sometimes, and would require very significant work to get something broken by design.

**Comment by scy** [comment]
I am by no means a beginner OSM user, but StreetComplete's behavior in this regard has led me to completely disable the "building type" quest.

While many users in this thread seem to have not come across `building:part` often, it's a very common thing where I live (rural Niedersachsen in Germany), and I suspect this is because the building shapes have been imported from government open data that way, i.e. separate ways for what are essentially parts of the same house.

* Giving these their own `building=*` types makes little sense, because then StreetComplete will start asking for their address.
* But merging the ways into one large building in a different editor isn't great either: The building parts will often have different heights, colors, roofs, or other properties, so it makes sense to leave them as separate ways, e.g. for additional detail for 3D renderers.
* Even if I tag them with `building:part=yes`, StreetComplete will _still_ ask what building type it is, which makes no sense whatsoever to me. (At least not until StreetComplete is able to deal with building parts better.)

I understand that being able to split buildings, or merge them, might be out of scope for StreetComplete. But the way it is right now is partially wrong, and at least confusing and irritating, even for experienced users.

My suggestion consists of two parts:

1. Do not request building type if the way is tagged as `building:part=yes`.
2. Allow giving "this is part of an adjacent, already tagged building" as an answer to the building type quest, which will then set `building:part=yes` on the way.

What do you think?

**Comment by matkoniecz** [comment]
> and I suspect this is because the building shapes have been imported from government open data that way

if you are in area with badly done import: I would open notes for such bogus data

> Even if I tag them with building:part=yes

are you removing invalid `building=yes` tag?

**Comment by matkoniecz** [comment]
> Allow giving "this is part of an adjacent, already tagged building" as an answer to the building type quest, which will then set building:part=yes on the way.

this is typically an insufficient fix - opening notes is better (and yes, it can be painful if you are in area with badly done import, some city in Poland had import that added each step in external staircases as separate tiny building...)

**Comment by scy** [comment]
> are you removing invalid `building=yes` tag?

Oh! I am not. (Well, to be exact, they have been tagged by other people.) The main part of the building is tagged `building=<whatever>`, the parts attached to it with `building:yes` _and_ `building:part=yes`, but there's no way around _all_ of the parts that's tagged with `building=<whatever>`.

Like, each way is tagged with `building=*`, while the non-main parts additionally have `building:part=yes`.

I see now what the issue is and will try re-tagging these as I come across them, thanks for the hint!

**Comment by matkoniecz** [comment]
> but there's no way around all of the parts that's tagged with building=<whatever>.

Ideally this one also be fixed and main building part also tagged as one more building part

Yes, if it is badly done import it will take a lot of work - but maybe it is possible to automate this? It may be worth asking on forums

Good news is that it affects not only SC, so fix will help not only here
